pkg/commands/options/publish.go

```diff
@@ -92,6 +92,11 @@ func AddPublishArg(cmd *cobra.Command, po *PublishOptions) {
 
 	cmd.Flags().BoolVarP(&po.PreserveImportPaths, "preserve-import-paths", "P", po.PreserveImportPaths,
 		"Whether to preserve the full import path after KO_DOCKER_REPO.")
+
+	// Set Base import Path if the KO_BASE_IMPORT_PATHS variable is set
+	if _, exists := os.LookupEnv("KO_BASE_IMPORT_PATHS"); exists {
+		po.BaseImportPaths = true
+	}
```

Comment:
I wonder if there is a way of doing this already via Viper (the config dual of cobra)?

Reply:
yep, i guess if we start doing this we may have switch to most commands using viper. if the ko maintainers are okay with this i will be happy to start this work.